inspirehep/modules/orcid/utils.py

```diff
@@ -0,0 +1,235 @@
+    user_works = api.read_record_member(
+        orcid,
+        'works',
+        oauth_token,
+        accept_type='application/orcid+json',
+    )
+
+    put_codes = [
+        str(item[0]) for item
+        in get_value(user_works, 'group.work-summary.put-code')
+    ]
+
+    # We can batch requests for _detailed_ records for maximum
+    # `WORKS_BULK_QUERY_LIMIT` put-codes at once:
+    detailed_works = {'bulk': []}
+
+    for put_code_batch in _split_lists(put_codes, WORKS_BULK_QUERY_LIMIT):
+        batch = api.read_record_member(
+            orcid,
+            'works',
+            oauth_token,
+            accept_type='application/orcid+json',
+            put_code=put_code_batch,
+        )
+
+        detailed_works['bulk'].extend(batch['bulk'])
+
+    # Now that we have all of the detailed records, we extract recids.
+    # If it's not possible (and it should always be), we put the put-code in
+    # `errors`
+    author_putcodes = []
+    errors = []
+
+    for item in detailed_works['bulk']:
+        put_code = get_value(item, 'work.put-code')
+        if not put_code:
+            continue
+
+        try:
+            url = get_value(item, 'work.url.value')
+            match = RECID_FROM_INSPIRE_URL.match(url)
+            recid = match.group(1)
+        except (TypeError, AttributeError):
+            errors.append(put_code)
+            continue
+
+        if put_code:
+            author_putcodes.append(
+                (recid, put_code)
+            )
+
+    return author_putcodes, errors
+
+
+def push_record_with_orcid(recid, orcid, oauth_token, put_code=None):
+    """Push record to ORCID with a specific ORCID ID.
+
+    Args:
+        recid (string): HEP record to push
+        orcid (string): ORCID identifier to push onto
+        oauth_token (string): ORCID user OAUTH token
+        put_code (Union[string, NoneType]): put-code to push record onto,
+            if None will push as a new record
+
+    Returns:
+        string: the put-code of the updated/inserted item
+    """
+    config = load_config()
+
+    server_name = config['SERVER_NAME']
+
+    record_api_endpoint = _get_api_url_for_recid(
+        server_name, config['SEARCH_UI_SEARCH_API'], recid
+    )
+
+    hep_response = requests.get(record_api_endpoint, headers={
+        'Accept': 'application/json'
+    })
+    record = hep_response.json()['metadata']
+
+    bibtex_response = requests.get(record_api_endpoint, headers={
+        'Accept': 'application/x-bibtex'
+    })
+    bibtex_string = bibtex_response.text
+
+    client_key = config['ORCID_APP_CREDENTIALS']['consumer_key']
+    client_secret = config['ORCID_APP_CREDENTIALS']['consumer_secret']
+
+    orcid_api = MemberAPI(client_key, client_secret)
+
+    orcid_xml = OrcidConverter(
+        record, server_name, put_code=put_code, bibtex_citation=bibtex_string
+    ).get_xml()
+
+    if put_code:
+        LOGGER.info("Pushing record #{} with put-code {}.".format(
```

Comment:
I would add the ORCID too, for an easy and quick check in case of errors :rocket: